Templates/Biostats_template.Rmd

```diff
@@ -30,6 +72,74 @@ output:
 ---
 ```
 
+Add this chunk to load libraries, define options and source the figure setting file.
+!!! HOW CAN I DISPLAY THE FULL CHUNK? !!!
+```{r setup-template, eval=FALSE, include=TRUE}
+#add all libraries that need loading
+library(learnr)
+
+#define options
+knitr::opts_chunk$set(echo = TRUE)
+
+#source figure settings
+source("set_theme_file.R")
+```
+
+
+
+# Text size, colours, etc
+
+Don't worry, the css styleguide file will take care of this.
+
+
+# Figures
+
+The set_theme_file.R will take care of colours, text size, theme etc in all the figures.
+This file has to be sourced in each Rmarkdown file (see Starting an .Rmd file above).
+
+The figures have the following settings:
+- theme_minimal()
+- text size: 12
+- colour: ... for discrete data use... and for continouse data use ...
+
+
+Each figure will have two chunks of code:
+- a chunk that displays the code, but does not do anything
+- a chunk that makes and prints the figure with the correct settings
+
+
+Here is an example:
+
+```{r make-figure, eval=FALSE}
+ggplot(penguins, aes(x = flipper_length_mm, y = bill_length_mm, colour = species)) +
+  geom_point() +
+  geom_smooth(method = "lm", formula = "y ~ x")
+
+
+```
+
+```{r display-figure, echo=FALSE, fig.cap="Figure 3.1 Penguine flipper and bill length."}
```

Comment:
If we are foing to number figures, probably ought to roll out fig references. - could refer to this figure as \@ref(fig:display-figure). Numbering is automatic. Need to use bookdown function in yaml. bookdown::html_document2 instead of html_document

Reply:
Ok, that works. Bookdown then also gives a number to each section, which is fine. But then it does not really work with the higher hierarchy that each section has a number (e.g. 1. working in R) and within these I suggest to use 1.1 getting started, 1.2 install R etc.
But there is probably a solution for that as well?

Reply:
You could do 
> (figure 1.backslash@ref(fig:my_figure)
Probably even better solutions within bookdown (it is expecting chapters etc). Problem we might have is that on page 2 of getting started, we will to start with figure 1.9 (or whatever) and then change this if the number of figures on page 1 changes. If we want to be able to refer to figures from other pages, then we need to do this, otherwise 1:n on each page is the simple solution

Reply:
yes, this is another thing we need to figure out. will ask the others what they think...